sydent/http/servlets/store_invite_servlet.py

```diff
@@ -147,8 +147,34 @@ def redact_email_address(self, address):
         # Extract strings from the address
         username, domain = address.split(u"@", 1)
 
-        # Obfuscate strings
-        redacted_username = self._redact(username, self.sydent.username_obfuscate_characters)
+        # Obfuscate the username portion
+
+        separator = self.sydent.third_party_invite_username_separator_string
+        if separator:
+            # If a separator string has been configured, we redact each component of the
+            # username individually.
+            redacted_username = ""
+
+            username_components = username.split(separator)
+            for index, component in enumerate(username_components):
+                if component:
+                    # Redact this component and append it to the final string
+                    redacted_username += (
+                        self._redact(component, self.sydent.username_obfuscate_characters)
+                    )
+
+                # Append the separator separately. This handles the case of multiple
+                # separators appearing sequentially in the username.
+                #
+                # The conditional here is to prevent a separator being appended to the end.
+                if index != len(username_components) - 1:
+                    redacted_username += separator
```

Comment:
I think this can be simplified, I'm not 100% sure it is identical, but I think it is.

```suggestion
            # Redact each component individually, if it has content. (No content implies multiple sequential separators.)
            redacted_components = [
                self._redact(component, self.sydent.username_obfuscate_characters) if component else "" for component in username.split(separator)
            ]
            redacted_username = separator.join(username_components)
```

Reply:
Hah, yep this works. And to think I was so close :) Thanks!